src-ui/src/app/data/filter-rule.ts

```diff
@@ -25,6 +25,34 @@ export function isFullTextFilterRule(filterRules: FilterRule[]): boolean {
   )
 }
 
+export function filterRulesDiffer(
+  filterRulesA: FilterRule[],
+  filterRulesB: FilterRule[]
+): boolean {
+  let modified = false
+  if (filterRulesA.length != filterRulesB.length) {
+    modified = true
+  } else {
+    modified = filterRulesA.some((rule) => {
+      return (
+        filterRulesB.find(
+          (fri) => fri.rule_type == rule.rule_type && fri.value == rule.value
+        ) == undefined
+      )
+    })
+
+    if (!modified) {
+      // only check other direction if we havent already determined is modified
+      modified = filterRulesB.some((rule) => {
```

Comment:
Why would this be necessary?

If we have two sets a and b of the same length and all elements of a are in b then all elements of b must be in a.

Or would it be possible to have duplicate filter rules in the list?

Reply:
Yea I had this thought. Not that I actually sat down to test it but I think youre right that this if will never be reached, unless there are dupes. Still seems no real cost to leaving it "just in case". This function has been around for a while and basically moved from elsewhere in this PR, I didnt modify the logic.

Reply:
Yea lets ditch it, good catch!

Reply:
Great, thanks!

It probably won't have a noticeable performance impact with few filter rules but in the default case where the rules did not change, it would have always compared twice.